ci/scripts/cpp_build.sh

```diff
@@ -118,12 +118,36 @@ if [ "${ARROW_USE_MESON:-OFF}" = "ON" ]; then
     fi
   }
 
+  ORIGINAL_CC="${CC}"
+  if [ -n "${CC}" ]; then
+    if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      CC="ccache ${CC}"
+    else
+      if command -v sccache; then
+        CC="sccache ${CC}"
+      fi
+    fi
+  fi
+
+  ORIGINAL_CXX="${CXX}"
+  if [ -n "${CXX}" ]; then
+    if [ "${ARROW_USE_CCACHE}" = "ON" ]; then
+      CXX="ccache ${CXX}"
+    else
+      if command -v sccache; then
+        CXX="sccache ${CXX}"
+      fi
+    fi
+  fi
   meson setup \
```

Comment:
Shouldn't we `export` those variables for them to be noticed by the meson process? Or is it just not necessary?

Reply:
My knowledge of bash syntax is pretty limited, but I did start with an export and was asked to remove in https://github.com/apache/arrow/pull/45858#discussion_r2002481071

Reply:
> My knowledge of bash syntax is pretty limited

Same for me... Apparently "being exported" is an attribute that is being set and inherited accross processes, though I don't see it specified clearly in the [POSIX doc](https://pubs.opengroup.org/onlinepubs/009695399/utilities/export.html). Is it bash-specific? @kou

Reply:
In section 2.12 of the Shell Command Language: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html


The shell execution environment is described as:

> Variables with the export attribute, along with those explicitly exported for the duration of the command, shall be passed to the utility environment variables

which works together with the docs you linked for the `export` special built-in command, that notes that the effect of `export XXX` is to mark the variable `XXX` with the export attribute.